File: Civitas-Java/Civitas/src/main/java/civitas/SorpresaPAGARCOBRAR.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 * @author elena
 * @date 29/11/2019
 */

public class SorpresaPAGARCOBRAR extends Sorpresa{
    
    SorpresaPAGARCOBRAR(int valor, String texto){   //Resto de sorpresas
        super(valor, texto);
    }
    
    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        Jugador jugador = todos.get(actual);
        
        if(jugadorCorrecto(actual, todos)){
            informe(actual, todos);
            jugador.modificarSaldo(valor);
        }
    }
    
    @Override
    public String toString(){
        return super.toString() + 
               "\nTipo Sorpresa: PAGARCOBRAR";
    }
}

File: Civitas-Ruby/lib/sorpresa_pagarcobrar.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative "diario"
require_relative "tipo_sorpresa"
require_relative "mazo_sorpresas"
require_relative "tablero"
require_relative "jugador"

module Civitas
  class SorpresaPagarcobrar < Sorpresa
    def initialize(valor, texto)
      super(valor, texto)
    end
    
    @override
    def aplicarajugador(actual, todos)
      if jugadorcorrecto(actual,todos)
        informe(actual, todos)
        todos[actual].modificasaldo(valor)
      end
    end
    
    @override
    def to_s
      super.to_s + 
      '\nTipo Sorpresa: PAGARCOBRAR'
    end
    
  end
end
